webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/CreateNodeData.scala

```diff
@@ -362,7 +378,10 @@ object Validation {
 
   def getMachine(id: NodeId, status: InventoryStatus, machineType: String): MachineInventory = {
     val machineId = MachineUuid(IdGenerator.md5Hash(id.value))
-    val machine   = machineType.toLowerCase
+    val machine   = machineType.toLowerCase match {
+      case "virtual" => "vm"
```

Comment:
this is to keep compatibility with old format where `vm` used to mean `Virtual, unknown provider`